### ISSUE TITLE
finish-args-unnecessary-xdg-data-access is incorrect

### ISSUE THREAD

**Issue by dusansimic** [flathub/flatpak-builder-lint#82]
The error means that the finish arg for xdg-data with a `:create` option at the end is not necessary since xdg dirs already exist. However, in this case I have an args that create subdirectories in the `xdg-data` dir.

```sh
--filesystem=xdg-data/backgrounds:create
--filesystem=xdg-data/gnome-background-properties:create
```

There should be no error for this as far as I understand.

**Comment by hfiguiere** [comment]
it is most of the time an error. so it is correct.

**Comment by barthalion** [comment]
If your app has a legit use of it, this is the place to report it. People tend not to know what they're doing and they blind-slap any permission that seems remotely related to the problem at hand.

**Comment by dusansimic** [comment]
> If your app has a legit use of it, this is the place to report it.

It does. My app relies on `$XDG_DATA_HOME/gnome-background-properties` and `$XDG_DATA_HOME/gnome-background-properties` to exist in order to work (it copies files into those directories). If they don't exist (which has happened in some environments https://github.com/dusansimic/dynamic-wallpaper/issues/5), the app breaks. It needs permission to both read and write those directories and create them if they don't exist. If there is a separate way to describe in a flatpak manifest what directory tree the app depends on, that would be a cleaner way to do so but this is the only one i found so far.

Also (this is not directly related to the issue at hand) I'd like to ask something while I'm here. If I'd like to add an exception, how do I go about doing that? I can't find any docs about it and from the source code I think I should make a `exceptions.json` file but I'm unsure how it should be structured.

**Comment by barthalion** [comment]
Please add your app's ID to https://github.com/flathub/flatpak-builder-lint/blob/master/flatpak_builder_lint/staticfiles/exceptions.json with a short explanation; as your app has been accepted before the linter has been introduced, it can be either "the app predates this linter rule" or technical explanation.

**Comment by dusansimic** [comment]
Done!

I'm just curious, is there a reason exceptions are centralized in such a way instead of each flatpak repo containing their own exceptions (for example, placing them in the the `flathub.json` or some other repo agnostic file). This way `flatpak-builder-lint` is tightly integrated into flathub and can't really be reliably used without modifications in some other flatpak repos.

**Comment by hfiguiere** [comment]
> I'm just curious, is there a reason exceptions are centralized in such a way instead of each flatpak repo containing their own exceptions

because otherwise rules can't be enforced.
